From 716b9fc3cdfb794590cd22df9ae6f5614feb2d9a Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 04 Oct 2018 14:17:35 +0300
Subject: [PATCH] release: add jenkins job parameters

Jenkins job expects for various parameters for the build.
Till now, the defaults were used and this has not covered
all the cases.
Prepare and provide the parameters for the Jenkins release job.

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/Makefile b/Makefile
index e3cf246..19dccc2 100644
--- a/Makefile
+++ b/Makefile
@@ -69,10 +69,13 @@
 
 RELEASE_REPOS := buildroot-external preboot u-boot linux armcp tbot fw_loader btl
 RELEASE_TAG_POINTER := latest-embedded-release
+RELEASE_BRANCH := $(shell git rev-parse --abbrev-ref HEAD)
+RELEASE_JOB_BRANCH := "package_branch=$(RELEASE_BRANCH)"
+RELEASE_VER_COMPAT := "release_version=$(patsubst v%,%,$(RELEASE_BRANCH))"
 RELEASE_JOB_URL := http://192.168.0.4:8080/job/goya-release-test/buildWithParameters
 RELEASE_JOB_CAUSE := "cause=Started_by_$(shell whoami)"
 RELEASE_JOB_TOKEN := "token=goya-release"
-RELEASE_JOB_PARAMS := $(RELEASE_JOB_TOKEN)&$(RELEASE_JOB_CAUSE)
+RELEASE_JOB_PARAMS := $(RELEASE_JOB_TOKEN)&$(RELEASE_JOB_CAUSE)&$(RELEASE_JOB_BRANCH)&$(RELEASE_VER_COMPAT)
 
 .PHONY: print-release-repos
 print-release-repos:
@@ -81,6 +84,9 @@
 .PHONY: gaudi-release goya-release
 gaudi-release goya-release: hl-check-release
 	$(eval HL_FULL_VER=hl-$(patsubst %-release,%,$@)-$(HL_RELEASE_VER))
+	$(eval RELEASE_JOB_PROD_NAME="product_name=$(patsubst %-release,%,$@)")
+	$(eval RELEASE_JOB_TAG_PTR="embedded_release_tag=$(HL_FULL_VER)")
+	$(eval RELEASE_JOB_PARAMS=$(RELEASE_JOB_PARAMS)&$(RELEASE_JOB_PROD_NAME)&$(RELEASE_JOB_TAG_PTR))
 	@repo forall $(RELEASE_REPOS) -c \
 		git tag -afm "Release Habanalabs $(HL_FULL_VER)" $(HL_FULL_VER)
 	@repo forall $(RELEASE_REPOS) -c \
